### PR TITLE
Create/update credentials on update and migrations

### LINKED ISSUE #4552: Remove raw credentials from cluster after edit
**User Story** It seems that we may have a bug related to cluster edition. After the cluster is created the credentials are stored in the `credentialsReference` and removed from other properties (like `token` for DigitalOcean). When user edits the cluster in the UI it seems that it is not happening. Please see: https://github.com/kubermatic/dashboard-v2/pull/1718#issuecomment-546233867.

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**:
 - Removes the projectID label during secret creation since that label isn't used anywhere.
 - Adds logic to update credentials in PatchEndpoint.
 - Adds code to create/update credentials for migration. This also changes the migration-related code to use controller-runtime.
 - Adds unit test for credentials for migration.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4552

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/crd/migrations/seed/migrations.go [hunk 1]
```diff
@@ -152,17 +154,29 @@ func setExposeStrategyIfEmpty(cluster *kubermaticv1.Cluster, ctx *cleanupContext
...
 
 func cleanupDashboardAddon(cluster *kubermaticv1.Cluster, ctx *cleanupContext) error {
-	if err := ctx.kubermaticClient.KubermaticV1().Addons(cluster.Status.NamespaceName).Delete("dashboard", &metav1.DeleteOptions{}); err != nil {
+	dashboardAddon := &kubermaticv1.Addon{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: cluster.Status.NamespaceName,
+			Name:      "dashboard",
+		},
+	}
+	if err := ctx.client.Delete(context.TODO(), dashboardAddon); err != nil {
```
Comment (maciaszczykm) [inline api/pkg/crd/migrations/seed/migrations.go]: You are using `context.TODO()` in a few places. Can this be modified to use existing context where possible, i.e. you have some context in this method as a param, would it work?

Comment (zreigz) [reply api/pkg/crd/migrations/seed/migrations.go]: good point, I've changed for the existing context

**Pull request merged**